orchestra/actions/install.py

```diff
@@ -70,36 +70,53 @@ def _run(self, args):
             os.path.relpath(self.config.installed_component_metadata_path(self.build.component.name), orchestra_root))
         new_files = [f for f in post_file_list if f not in pre_file_list]
 
-        if not args.no_merge:
+        if not cmdline_args.no_merge:
             if is_installed(self.config, self.build.component.name):
                 logger.info("Uninstalling previously installed build")
                 uninstall(self.build.component.name, self.config)
 
             logger.info("Merging installed files into orchestra root directory")
             self._merge()
 
-            # Write file metadata and index
-            os.makedirs(self.config.installed_component_metadata_dir(), exist_ok=True)
-            metadata = {
-                "component_name": self.build.component.name,
-                "build_name": self.build.name,
-                "install_time": install_end_time - install_start_time,
-                "source": source,
-                "self_hash": self.build.component.self_hash,
-                "recursive_hash": self.build.component.recursive_hash,
-                "binary_archive_path": os.path.join(self.build.binary_archive_dir, self.build.binary_archive_filename),
-            }
-            with open(self.config.installed_component_metadata_path(self.build.component.name), "w") as f:
-                json.dump(metadata, f)
-
-            with open(self.config.installed_component_file_list_path(self.build.component.name), "w") as f:
-                new_files = [f"{f}\n" for f in new_files]
-                f.writelines(new_files)
+            self._update_metadata(new_files,
+                                  install_end_time - install_start_time,
+                                  source,
+                                  set_manually_installed)
 
-        if not args.keep_tmproot:
+        if not cmdline_args.keep_tmproot:
             logger.info("Cleaning up tmproot")
             self._cleanup_tmproot()
 
+    def _update_metadata(self, file_list, install_time, source, set_manually_insalled):
+        # Write file metadata and index
+        os.makedirs(self.config.installed_component_metadata_dir(), exist_ok=True)
+
+        try:
+            with open(self.config.installed_component_metadata_path(self.build.component.name), "w") as f:
+                metadata = json.load(f)
+        except (IOError, json.JSONDecodeError):
+            metadata = {}
```

Comment:
What about `if os.path.exists(...)` instead of suppressing errors?
If you prefer it like this we should at least emit a warning, don't you think?
Minor thing anyway.